app/search/controllers.py

```diff
@@ -46,7 +46,14 @@ def search_packages():
         q = request.args.get('q')
         if q is None:
             q = ''
-        result = DataPackageQuery(query_string=q.strip()).get_data(20)
+        limit = request.args.get('limit')
+        try:
+            limit = int(limit)
```

Comment:
Surely this should be:

```
limit = min(int(limit), 1000)
```

I also think the limit processing stuff should be in DataPackageQuery so it is common to all code paths.